src/main/java/org/veupathdb/service/eda/compute/plugins/correlationassaymetadata/CorrelationAssayMetadataPlugin.java

```diff
@@ -0,0 +1,151 @@
+
+  private static final String INPUT_DATA = "correlation_input";
+
+  public CorrelationAssayMetadataPlugin(@NotNull PluginContext<CorrelationPluginRequest, CorrelationComputeConfig> context) {
+    super(context);
+  }
+
+  @NotNull
+  @Override
+  public List<StreamSpec> getStreamSpecs() {
+    // Get the collection variable and its entity
+    CorrelationComputeConfig computeConfig = getConfig();
+    CollectionSpec collectionVariable1 = computeConfig.getCollectionVariable1();
+    String entityId = collectionVariable1.getEntityId();
+
+    // Wrangle into correct types for what follows
+    EntityDef entity = getContext().getReferenceMetadata().getEntity(entityId).orElseThrow();
+    VariableSpec collectionVariable1VarSpec = VariableDef.newVariableSpec(entityId, collectionVariable1.getCollectionId());
+
+
+    // Grab all continuous variabls from ancestors
+    // The next line only grabs the second-up ancestor instead of all of them, because when i grabbed them all
+    // the variables wold be duplicated in the output. I tried distinct() but that didn't work. Any ideas?
+    // I also noticed we were receiving stable ids in the oriringal request, so i've filtered them out below. There 
+    // is probably a nicer way to do this.
+    // This at least works for now!
+    Stream<VariableDef> descendantVariableStream = getContext().getReferenceMetadata().getAncestors(entity).get(1).getVariables().stream();
+    List<VariableDef> metadataVariables = descendantVariableStream.filter(var -> var.getDataShape() == APIVariableDataShape.CONTINUOUS).filter(var -> !var.getVariableId().contains("_stable_id")).toList();
+
+
+    return List.of(new StreamSpec(INPUT_DATA, getConfig().getCollectionVariable1().getEntityId())
+        .addVars(getUtil().getChildrenVariables(collectionVariable1VarSpec))
+        .addVars(metadataVariables)
+      );
+  }
+
+  @Override
+  protected void execute() {
+
+    CorrelationComputeConfig computeConfig = getConfig();
+    PluginUtil util = getUtil();
+    ReferenceMetadata meta = getContext().getReferenceMetadata();
+
+    // Get compute parameters
+    String method = computeConfig.getCorrelationMethod().getValue();
+    CollectionSpec collectionVariable1 = computeConfig.getCollectionVariable1();
+    String entityId = collectionVariable1.getEntityId();
+
+    // Wrangle into helpful types
+    EntityDef entity = meta.getEntity(entityId).orElseThrow();
+    VariableDef computeEntityIdVarSpec = util.getEntityIdVarSpec(entityId);
+    String computeEntityIdColName = util.toColNameOrEmpty(computeEntityIdVarSpec);
+    VariableSpec collectionVariable1VarSpec = VariableDef.newVariableSpec(entityId, collectionVariable1.getCollectionId());
+
+    // Get record id columns
+    List<VariableDef> idColumns = new ArrayList<>();
+    for (EntityDef ancestor : meta.getAncestors(entity)) {
+      idColumns.add(ancestor.getIdColumnDef());
+    }
```

Comment:
im a little surprised this is the best way we have to do this. seems an operation wed like to do often enough itd warrant a helper as well..

Reply:
https://github.com/VEuPathDB/service-eda-compute/issues/73